### DISCUSSION TITLE
Identifying which sensor is which

### DISCUSSION THREAD

**Discussion by Oscar-Geare** [telekom-security/tpotce#1646] [Q&A]
Hello,

I have a hive installation with sensors in different network segments. I have cowrie deployed to them, and I'm looking on Kibana at the data. I see there is a "sensor" data field which gives me an ID. How can I associate this with the sensors I have deployed?

Additionally, for both sensors the tpot_hostname field bot appears as the Hive hostname. Is this intended or have I messed something up?

**Comment by dmille6** [comment]
you should have a "hostname" in the kibana name, and a source/destination IP. between those 3, you should be able to determine which sensor is which

**Reply by Oscar-Geare** [reply]
I'm not sure if it's because of something I've done, but I'm looking at the following fields in elastic and not seeing anything to correlate on.

t-pot_ip_ext = external IP of the Hive
t-pot_ip_int = internal IP of the hive
t-pot hostname = hostname of the hive
sensor = alphanumeric value, there are two unique values which is to be expected as there are two sensors
host = alphanumeric value, different to the sensor. None of these relate to the hostname of the sensors.
src_ip = ip of the attack event.

**Reply by dmille6** [reply]
go to kibwana -> discover.. open up one of the log entries.. (make sure you have 'rows per page is set to 100')
look for 'T-Pot Hostname'

**Reply by Oscar-Geare** [reply]
Yes that only shows the hive hostname for me.

**Reply by dmille6** [reply]
just tested it.. you're right.. i think this is a bug. @t3chn0m4g3 ... what do ya think.. ?

**Reply by t3chn0m4g3** [reply]
Just setup two nodes, one as hive named `hive` the other as sensor named `sensor`.

Everything works as intended...
![image](https://github.com/user-attachments/assets/041ef6f0-d6f7-4212-b4f9-7df5acdc3020)

**Reply by Oscar-Geare** [reply]
Ok, thanks for checking it out. I wonder what dmille6 and I have done which means it doesn't work for us. I'll deploy again and try to see what is going on.

**Reply by t3chn0m4g3** [reply]
@Oscar-Geare Either the sensor is not even transmitting logs, check on sensor with `docker logs logstash -f` while accessing i.e. Cowrie or it has a wrong `tpotce/docker-compose.yml` which has to say `SENSOR` on the top (which should not happen, when using `deploy.sh`).
Or there is a certificate mismatch (briefly check with setting `LS_SSL_VERIFICATION=none` in the `.env` on the sensor) and follow the ReadMe for a full explanation.

**Reply by Oscar-Geare** [reply]
Thanks, I think I missed the step with modifying the certificate as I'm getting the below errors:

[ERROR] 2024-09-04 05:04:58.422 [[http_output]>worker1] http - Could not fetch URL {:url=>"https://<HIVE_IP_EXTERNAL>:64294", :method=>:post, :message=>"Certificate for <HIVE_IP_EXTERNAL> doesn't match any of the subject alternative names: [<HIVE_IP_INTERNAL>]", :class=>Manticore::UnknownException, :will_retry=>false}

To confirm, if I don't have a publicly signed certificate I'll need to edit the config to read as so?

openssl req \
    -nodes \
    -x509 \
    -sha512 \
    -newkey rsa:8192 \
    -keyout "nginx.key" \
    -out "nginx.crt" \
    -days 3650 \
    -subj '/C=AU/ST=Some-State/O=Internet Widgits Pty Ltd' \
    -addext "subjectAltName = IP:<HIVE_IP>, IP:<SENSOR_1>, IP:<SENSOR_2>"

**Reply by t3chn0m4g3** [reply]
Yes, full guide [here](https://github.com/telekom-security/tpotce?tab=readme-ov-file#planning-and-certificates).

**Reply by Oscar-Geare** [reply]
Thank you. I thought it was unnecessary when deploying as I do not need the HIVE to be contacted on multiple IP Addresses, only one.

**Reply by t3chn0m4g3** [reply]
Did that IP change at some point or is a NAT involved?

**Reply by Oscar-Geare** [reply]
No, the IP has not changed. However, the resource is deployed on Azure as I've just been testing as a POC. It should have its own static, public, IP address.

**Reply by t3chn0m4g3** [reply]
Thanks, hope everything works now :bowtie:

**Reply by dmille6** [reply]
@t3chn0m4g3 thanks for the quick response and spending the time to build out a test. I'll dig in deeper on my end. i thew something together too.. probably got the cert wrong so I'm only getting stuff from the hive server.
